registration-checker-compiler-plugin/src/test/resources/serializers/CorrectSerializer.scala

```diff
@@ -1,8 +1,9 @@
 package org.random.project
 
-import org.virtuslab.ash.Serializer
+import org.virtuslab.ash.annotation.{Serializer, SerializerRegexFilter}
 
 @Serializer(classOf[SerializableTrait])
+@SerializerRegexFilter(".*Option.*")
 class CorrectSerializer {
```

Comment:
Also! a point valid for Hydra: will the plugin work as expected when the `Register`s are spread across multiple classes? As suggested on a recent Hydra PR (one class with codecs per package aka _module_)

Reply:
Suppose there are three traits and marker trait:
```scala
@SerializablilityTrait
trait Mark

sealed trait One extends Mark
sealed trait Two extends Mark
sealed trait Three extends Mark
```

This will compile correctly:

```scala
trait Codecs{
  implicit val codec = ???
}

@Serializer
class Serializer extends Codecs {
  lazy val codecs = Seq(Register[One], Register[Two], Register[Three])
}
```

While this will fail:

```scala
trait Registers{
 lazy val register = Seq(Register[One], Register[Two], Register[Three])
}

@Serializer
class Serializer extends Codecs {
  lazy val codecs = this.register
}
```

Reply:
If there are some in between traits, for example:

```scala
@SerializablilityTrait
trait Mark
trait Trans

sealed trait One extends Trans
sealed trait Two extends Trans
sealed trait Three extends Mark
```

Then this compiles:
```scala
@Serializer
class Serializer{
  lazy val codecs = Seq(Register[Trans], Register[Three])
}
```

This fails:
```scala
@Serializer
class Serializer{
  lazy val codecs = Seq(Register[One], Register[Two], Register[Three])
}
```